pandas/core/array_algos/transforms.py

```diff
@@ -10,6 +10,10 @@
 def shift(values: np.ndarray, periods: int, axis: int, fill_value) -> np.ndarray:
     new_values = values
 
+    if periods == 0:
+        # TODO: should we copy here?
+        return new_values
```

Comment:
I think we should, do we elsewhere?

Reply:
looks like we currently do a copy in 1/2 of places where this is called